packages/gatsby-transformer-remark/src/utils/generate-graphql-type.js

```diff
@@ -0,0 +1,27 @@
+const { get } = require(`lodash`)
+
+module.exports = async function generateGraphQLType(
+  helpers,
+  pluginOptions,
+  frontmatter
+) {
+  const { reporter } = helpers
+  const { contentTypes } = pluginOptions
+
+  if (contentTypes) {
+    const type =
+      typeof contentTypes.resolveType === `string`
+        ? get(frontmatter, contentTypes.resolveType)
+        : await contentTypes.resolveType(frontmatter)
+
+    if (!contentTypes.types.includes(type)) {
+      reporter.error(
+        `Type '${type}' does not match any of the provided contentTypes.`
```

Comment:
Can you mention the configuration settings here so that a user can know exactly where to go to fix this?

Reply:
Absolutely!